docs: Add examples of building and publishing images (#108)
apple/swift-container-plugin: Sources/swift-container-plugin/Documentation.docc/run.md

```
% curl localhost:8080
Hello World, from Hummingbird on Ubuntu 24.04.2 LTS
```

### Build and run in one step

Swift Container Plugin prints a reference to the newly built image on standard output.

> The image reference can then be piped to the deployment command, or passed as an argument using shell output substitution.

Reviewer: ```suggestion
You can pipe the image reference to a deployment command or pass it as an argument using shell output substitution.
```

> This allows a container image to be built and deployed in one shell command, using a convenient pattern offered by tools such as [ko.build](https://ko.build):

Reviewer: ```suggestion
This allows you build and deploy a container image in one shell command, using a convenient pattern offered by tools such as [ko.build](https://ko.build):
```
